OpenRA.Mods.Common/Traits/Cloak.cs

```diff
@@ -101,7 +103,11 @@ void INotifyCreated.Created(Actor self)
 
 		void INotifyDamage.Damaged(Actor self, AttackInfo e)
 		{
-			if (Info.UncloakOn.HasFlag(UncloakType.Damage))
+			if (e.Damage.Value == 0)
+				return;
+			var isHeal = e.Damage.Value < 0;
+			var isSelfHeal = isHeal && e.Attacker == self;
+			if (Info.UncloakOn.HasFlag(isSelfHeal ? UncloakType.SelfHeal : isHeal ? UncloakType.Heal : UncloakType.Damage))
```

Comment:
Having a nested terniary statement inline as a function arg makes me a bit squeemish.  Can you pull this out into its own variable?  If you check `isHeal` before `isSelfHeal` you can also simplify the latter down to `e.Attacker == self`.